opentok-whiteboard.js

```diff
@@ -236,21 +217,47 @@ var OpenTokWhiteboard = angular.module('opentok-whiteboard', ['opentok'])
                 
                 var offset = angular.element(canvas).offset(),
                     scaleX = canvas.width / element.width(),
-                    scaleY = canvas.height / element.height();
-                var offsetX, offsetY, x, y;
+                    scaleY = canvas.height / element.height(),
+                    offsetX = event.offsetX || event.originalEvent.pageX - offset.left ||
+                       event.originalEvent.touches[0].pageX - offset.left,
+                    offsetY = event.offsetY || event.originalEvent.pageY - offset.top ||
+                       event.originalEvent.touches[0].pageY - offset.top,
+                    x = offsetX * scaleX,
+                    y = offsetY * scaleY,
+										mode = !scope.erasing ? "eraser" : "pen",
+                    start;
                 
                 switch (event.type) {
                 case 'mousedown':
                 case 'touchstart':
-                    offsetX = event.offsetX || event.originalEvent.pageX - offset.left ||
-                        event.originalEvent.touches[0].pageX - offset.left,
-                    offsetY = event.offsetY || event.originalEvent.pageY - offset.top ||
-                        event.originalEvent.touches[0].pageY - offset.top,
-                    x = offsetX * scaleX,
-                    y = offsetY * scaleY;
+                    // Start dragging
                     client.dragging = true;
+                    
+                    // Where the line
                     client.lastX = x;
                     client.lastY = y;
+                    
+                    if( window.path ) {
+                        window.path.selected = false;
+                    }
+                    
+                    // Create a new path object
+                    window.path = new paper.Path();
```

Comment:
The `window.path()` doesn't get created until the current user starts drawing. This means that if you receive the events from someone else drawing then you get an error about window.path being undefined.